openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -112,18 +162,60 @@ def _check_for_pacing_changes(previous_course_overview, updated_course_overview)
 
 
 def _check_for_cert_availability_date_changes(previous_course_overview, updated_course_overview):
-    """ Checks if the cert available date has changed and if so, sends a COURSE_CERT_DATE_CHANGE signal"""
-    if previous_course_overview.certificate_available_date != updated_course_overview.certificate_available_date:
+    """
+    Checks if the certificate available date (CAD) or the certificates display behavior (CDB) of a course run has
+    changed during a course overview update. If so, we emit the COURSE_CERT_DATE_CHANGE signal to ensure other parts of
+    the system are aware of the change. The `credentials` app listens for this signal in order to keep our certificate
+    visibility settings in the Credentials IDA up to date.
+
+    Args:
+        previous_course_overview (CourseOverview): the current course overview instance for a particular course run
+        updated_course_overview (CourseOverview): an updated course overview instance, reflecting the current state of
+            data from the modulestore/Mongo
+
+    Returns:
+        None
+    """
+    def _send_course_cert_date_change_signal():
+        """
+        A callback used to fire the COURSE_CERT_DATE_CHANGE Django signal *after* the ORM has successfully commit the
+        update.
+        """
+        COURSE_CERT_DATE_CHANGE.send_robust(sender=None, course_key=str(updated_course_overview.id))
+
+    course_run_id = str(updated_course_overview.id)
+    prev_available_date = previous_course_overview.certificate_available_date
+    prev_display_behavior = previous_course_overview.certificates_display_behavior
+    prev_end_date = previous_course_overview.end  # `end_date` is a deprecated field, use `end` instead
+    updated_available_date = updated_course_overview.certificate_available_date
+    updated_display_behavior = updated_course_overview.certificates_display_behavior
+    updated_end_date = updated_course_overview.end  # `end_date` is a deprecated field, use `end` instead
+
+    send_signal = False
+    if prev_available_date != updated_available_date:
         LOG.info(
-            f"Certificate availability date for {str(updated_course_overview.id)} has changed from " +
-            f"{previous_course_overview.certificate_available_date} to " +
-            f"{updated_course_overview.certificate_available_date}. Sending COURSE_CERT_DATE_CHANGE signal."
+            f"The certificate available date for {course_run_id} has changed from {prev_available_date} to "
+            f"{updated_available_date}. Firing COURSE_CERT_DATE_CHANGE signal."
         )
+        send_signal = True
+    elif prev_display_behavior != updated_display_behavior:
```

Comment:
is it correct for these to be `elif`?  From a pure functionality standpoint, `send_signal = True` on any of them, but do you not want the log message to fire if multiple conditions are true?

Reply:
Ooo! I think this is a great point. I was being too rigid in my thinking. Let me fix this up.